python/ray/data/random_access_dataset.py

```diff
@@ -209,7 +210,19 @@ def get(self, block_index, key):
 
     def multiget(self, block_indices, keys):
         start = time.perf_counter()
-        result = [self._get(i, k) for i, k in zip(block_indices, keys)]
+        if self.dataset_format == "arrow" and len(set(block_indices)) == 1:
+            # Fast path: use np.searchsorted for vectorized search on a single block.
+            # This is ~3x faster than the naive case.
+            block = self.blocks[block_indices[0]]
+            col = block[self.key_field]
+            indices = np.searchsorted(col, keys)
+            acc = BlockAccessor.for_block(block)
+            result = [
+                acc._create_table_row(acc.slice(i, i + 1, copy=True)) for i in indices
+            ]
+            # assert result == [self._get(i, k) for i, k in zip(block_indices, keys)]
```

Comment:
Remove commented assert, unless you're wanting to keep it for documentation purposes?
```suggestion
            # assert result == [self._get(i, k) for i, k in zip(block_indices, keys)]
```

Reply:
Yup, keeping it around in case someone wants to debug in the future.